test-suite/Makefile

```diff
@@ -690,7 +690,7 @@ ide : $(patsubst %.fake,%.fake.log,$(wildcard ide/*.fake))
 	@echo "TEST      $<"
 	$(HIDE){ \
 	  echo $(call log_intro,$<); \
-	  $(BIN)fake_ide "$<" "-q -async-proofs on -async-proofs-tactic-error-resilience off -async-proofs-command-error-resilience off $(call get_coq_prog_args,"$<")" 2>&1; \
+	  ./fake_ide.exe "$(BIN)" "$<" "-q -async-proofs on -async-proofs-tactic-error-resilience off -async-proofs-command-error-resilience off $(call get_coq_prog_args,"$<")" 2>&1; \
```

Comment:
other parts of this makefile do stuff like `export coqc="eval $(coqc)"` to pass the path of coq binaries.

I think it is OK to pass the path of coqidetop to fake_ide, but I think you should pass the whole path, eg define a coqidetop variable in the Makefile and pass its value (akin to coqc). It should let you avoid to expose System.coq_executable_extension in the first place, IMO.

Reply:
So fake_ide would read the environment variable?

Reply:
No, I was suggesting to have a Make variable contain the full path of coqidetop, like we do for coqc
https://github.com/coq/coq/blob/f684f34eed7889cbced249630e4ffe9306b1401b/test-suite/Makefile#L71
and pass that value to `./fake_ide.exe`. In that was you should not need to craft that path in fake_ide, you can just use the argument as is. This, in turn, will make it unnecessary to expose System.coq_executable_extension.

Reply:
Ok, now I got it.